core/arch/instr_inline.h

```diff
@@ -276,10 +276,14 @@ opnd_create_pc(app_pc pc)
 #    define OPND_GET_BASE(opnd) (GET_BASE_DISP(opnd).base_reg)
 #    define OPND_GET_DISP(opnd) (GET_BASE_DISP(opnd).disp)
 #    ifdef X86
-#        define OPND_GET_INDEX(opnd)                                \
-            (GET_BASE_DISP(opnd).index_reg_is_zmm                   \
-                 ? DR_REG_START_ZMM + GET_BASE_DISP(opnd).index_reg \
-                 : GET_BASE_DISP(opnd).index_reg)
+#        define OPND_GET_INDEX(opnd)                                                     \
+            ({                                                                           \
```

Comment:
Use do..while(0) like all our other multi-statement macros do: it is considered the superior approach, allowing for a trailing ; (see various stackoverflow, etc. discussions).

Reply:
But this doesn't work with a return value, since I need to return something. I need a statement expression, but on the other hand the Windows compiler didn't like it (see AppVeyor below). Will do inline function instead.

Reply:
The opnd_t struct has historically blocked inlining by multiple compilers, which is why we continue to use macros around this type.

Reply:
Done.

Reply:
Reverted to original.